pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherSingleActiveConsumer.java

```diff
@@ -136,6 +139,24 @@ protected boolean pickAndScheduleActiveConsumer() {
         }
     }
 
+    private int peekConsumerIndexFromHashRing(NavigableMap<Integer, Integer> hashRing) {
+        int hash = Murmur3Hash32.getInstance().makeHash(topicName);
+        Map.Entry<Integer, Integer> ceilingEntry = hashRing.ceilingEntry(hash);
+        return ceilingEntry != null ? ceilingEntry.getValue() : hashRing.firstEntry().getValue();
+    }
+
+    private NavigableMap<Integer, Integer> makeHashRing(int consumerSize) {
+        NavigableMap<Integer, Integer> hashRing = new TreeMap<>();
+        for (int i = 0; i < consumerSize; i++) {
+            for (int j = 0; j < CONSUMER_CONSISTENT_HASH_REPLICAS; j++) {
+                String key = consumers.get(i).consumerName() + j;
```

Comment:
We can make it configurable in the future. And the key_shared subscription also uses 100 replicas. So I apply the same value here.

Reply:
We need to look at or do some testing to verify if there is more impact in bad conditions (many failover consumers). Maybe it's not a critical problem in the real use case.

Reply:
I understand your point. I'm also thinking about it to avoid building the hash ring for each consumer adding/removing. But the issue is a higher priority consumer might be added. Add maintaining a hash ring is also memory resource consumption.

I will do a test later. In most cases, consumers should be under hundreds.